### ISSUE TITLE
model.predict doesn't read some filenames

### ISSUE THREAD

**Issue by nadinator** [ultralytics/ultralytics#4516]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection

### Bug

When you run `model.predict(filename)` on a file who's name either a) Contains non-utf8 characters or b) contains spaces, you run into a FileNotFound error. This isn't necessarily a bug (just a lack of a certain feature), but in that case the error should in my view be changed to something like NotImplemented.

### Environment

- Ultralytics 8.0.155
- Python 3.11.4
- Windows 10
- Torch 2.0.1+cpu

### First example
```python
from ultralytics import YOLO

filename_with_spaces = 'filename with spaces.jpg'
model = YOLO('yolov8n.pt')
model.predict(filename_with_spaces)
```

### Second example
```python
from ultralytics import YOLO

filename_with_non_utf8 = 'veículo.jpg'
model = YOLO('yolov8n.pt')
model.predict(filename_with_non_utf8)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@nadinator inference on images with spaces works correctly for me:

<img width="1149" alt="Screenshot 2023-08-23 at 14 27 33" src="https://github.com/ultralytics/ultralytics/assets/26833433/4b3af2a9-a27c-43f3-9871-a9983784f73f">

**Comment by glenn-jocher** [comment]
@nadinator I've opened up #4517 to add these two edge cases to our tests. They're passing locally for me on macos:

<img width="1381" alt="Screenshot 2023-08-23 at 14 37 58" src="https://github.com/ultralytics/ultralytics/assets/26833433/9dab10f7-04fe-4cd4-9cc8-e571a4bf1119">

**Comment by nadinator** [comment]
Might be a windows error in that case.

**Comment by glenn-jocher** [comment]
@nadinator thank you for your response. It's possible that the bug you've encountered might be exclusive to the Windows environment. We will investigate this issue further to confirm if this is the case. 

Please note, Ultralytics YOLOv8 supports all popular operating systems and we strive to ensure that functionalities work across all of them. We appreciate your patience and support. Your feedback is vital to making YOLOv8 better for everyone.

**Comment by glenn-jocher** [comment]
@nadinator I'm working on some fixes in https://github.com/ultralytics/ultralytics/pull/4524 that should address this in windows. I'm running extra Windows CI in https://github.com/glenn-jocher/ultralytics/actions/runs/5956325696/job/16156846115?pr=13 on these changes in my personal branch.

**Comment by glenn-jocher** [comment]
Great news 😃! Your original issue may now be resolved ✅ in PR #4524. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by nadinator** [comment]
It's fixed, thanks! Both cases too. May I ask how you solved this problem?

**Comment by glenn-jocher** [comment]
@nadinator i'm glad to hear that the issue has been resolved!

To fix the problem, we made changes to how YOLOv8 processes file paths. It now correctly handles file paths with non-UTF8 characters and spaces. This was done by updating our file path encoding and decoding processes to ensure compatibility across different systems and file types.

Please let us know if you encounter any other issues. We're here to help! Happy training with YOLOv8.

**Comment by nadinator** [comment]
This tests non-utf characters on inference, but not in training, where I still get errors of the same kind. Any plans to fix this?

**Comment by glenn-jocher** [comment]
@nadinator absolutely, we're addressing this to ensure training processes support non-UTF8 characters as smoothly as inference now does. Thank you for pointing it out! The team is on it and we'll keep you updated on our progress. 🚀 In the meantime, a temporary workaround could be to ensure your training dataset paths are UTF-8 compatible. Thanks again for your feedback!
